nextflow.config

```diff
@@ -3,131 +3,155 @@
   // So -60 would allow 10 mismatches or ~ 8 x 1-2bp indels
   // Bismark default is 0.2 (L,0,-0.2), Bowtie2 default is 0.6 (L,0,-0.6)
-  mindepth = 0
-  ignoreFlags = false
+
+  // Boilerplate options
+  name = false
+  multiqc_config = "$baseDir/assets/multiqc_config.yaml"
+  email = false
+  email_on_fail = false
+  max_multiqc_email_size = 25.MB
+  plaintext_email = false
+  monochrome_logs = false
+  help = false
+  params.project = false
+  params.clusterOptions = false
+  igenomes_base = "s3://ngi-igenomes/igenomes/"
+  tracedir = "${params.outdir}/pipeline_info"
+  awsqueue = false
+  awsregion = 'eu-west-1'
+  igenomes_ignore = false
+  custom_config_version = 'master'
+  custom_config_base = "https://raw.githubusercontent.com/nf-core/configs/${params.custom_config_version}"
+  hostnames = false
+  config_profile_description = false
+  config_profile_contact = false
+  config_profile_url = false
 }
 
-profiles {
+// Container slug. Stable releases should specify release tag!
+// Developmental code should specify :dev
+process.container = 'nfcore/methylseq:1.4'
 
-  standard {
-    includeConfig 'conf/base.config'
-    includeConfig 'conf/igenomes.config'
-  }
+// Load base.config by default for all pipelines
+includeConfig 'conf/base.config'
+
+// Load nf-core custom profiles from different Institutions
+try {
+  includeConfig "${params.custom_config_base}/nfcore_custom.config"
+} catch (Exception e) {
+  System.err.println("WARNING: Could not load nf-core/config profiles: ${params.custom_config_base}/nfcore_custom.config")
+}
+
+profiles {
+  aws { includeConfig 'conf/aws.config' }
   conda { process.conda = "$baseDir/environment.yml" }
+  debug { process.beforeScript = 'echo $HOSTNAME' }
   docker { docker.enabled = true }
-  singularity { singularity.enabled = true }
-  uppmax {
-    includeConfig 'conf/base.config'
-    includeConfig 'conf/uppmax.config'
-    includeConfig 'conf/igenomes.config'
-  }
-  uppmax_devel {
-    includeConfig 'conf/base.config'
-    includeConfig 'conf/uppmax.config'
-    includeConfig 'conf/uppmax-devel.config'
-    includeConfig 'conf/igenomes.config'
-  }
-  aws {
-    includeConfig 'conf/base.config'
-    includeConfig 'conf/aws.config'
-    includeConfig 'conf/igenomes.config'
-  }
-  test {
-    includeConfig 'conf/base.config'
-    includeConfig 'conf/test.config'
-  }
-  none {
-    // Don't load any config (for use with custom home configs)
-  }
-
+  singularity { singularity.enabled = true
+                singularity.autoMounts = true }
+  test { includeConfig 'conf/test.config' }
 }
 
+// Avoid this error:
+// WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
+// Testing this in nf-core after discussion here https://github.com/nf-core/tools/pull/351, once this is established and works well, nextflow might implement this behavior as new default.
+docker.runOptions = '-u \$(id -u):\$(id -g)'
```

Comment:
Shouldn't this be specified in the docker profile?

Reply:
Agreed 👍